os/fs/driver/mtd/smart.c

```diff
@@ -4894,11 +4905,23 @@ static int smart_journal_move_to_next(FAR struct smart_struct_s *dev)
 	/* Check we have to move to next block, delete current block if it is full */
 	if (((old_seq + 1) % dev->njournalPerBlk) == 0) {
 		old_block = old_seq / dev->njournalPerBlk;
-		ret = smart_journal_erase(dev, old_block + dev->neraseblocks);
+		old_block += dev->neraseblocks;
```

Comment:
Could you clarify why this is added?

Reply:
If you meaning dev->neraseblocks, then dev->neraseblocks is MTD block driver's area.
From dev->neraseblocks, journal area started.
To get entire of eraseblocks that is allocated to smartfs, you can use dev->geo.eraseblocks.